packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -206,77 +196,43 @@ async function instantiateModule(
   return Object.freeze(ssrModule)
 }
 
+const isDedupable = (id: string) =>
+  bareImportRE.test(id) && !(isBuiltin(id) || id.endsWith('.node'))
+
 // In node@12+ we can use dynamic import to load CJS and ESM
 async function nodeImport(
   id: string,
   importer: string,
-  resolveOptions: InternalResolveOptions
+  dedupe?: string[],
+  projectRoot?: string
 ) {
-  // Node's module resolution is hi-jacked so Vite can ensure the
-  // configured `resolve.dedupe` and `mode` options are respected.
-  const viteResolve = (
-    id: string,
-    importer: string,
-    options = resolveOptions
-  ) => {
-    const resolved = tryNodeResolve(id, importer, options, false)
-    if (!resolved) {
-      const err: any = new Error(
-        `Cannot find module '${id}' imported from '${importer}'`
-      )
-      err.code = 'ERR_MODULE_NOT_FOUND'
-      throw err
-    }
-    return resolved.id
-  }
+  let unhookNodeResolve: (() => void) | undefined
+  if (dedupe && projectRoot) {
+    const resolve = Module.createRequire(projectRoot + '/index.js')
```

Comment:
where does this magic /index.js come from?

Reply:
The `createRequire` call needs a file in the project root, but it doesn't have to be a real file.